### ISSUE TITLE
ZFS Support

### ISSUE THREAD

**Issue by gabeduke** [rancher/k3c#14]
K3c does not honor the default plugin config in  `/etc/containerd/config.toml`. 

> **Edit**: Since writing this I've realized the default containerd config is not used. It would be great have an option to pass a snapshotter config so k3s can run on zfs

---

`k3c daemon`
```sh
INFO[2020-01-30T21:34:20.769021975-05:00] Start cri plugin with config {PluginConfig:{ContainerdConfig:{Snapshotter:overlayfs [...]
```

`/etc/containerd/config.toml`
```yaml
 [...]
    [plugins."io.containerd.grpc.v1.cri".containerd]
      snapshotter = "zfs"
      default_runtime_name = "runc"
      no_pivot = false
 [...]
```

**Comment by dweomer** [comment]
@gabeduke k3c currently runs with an embedded **containerd** with it's own config emitted at `/var/lib/rancher/k3c/etc/containerd/config.toml` by default ... edits to which are overwritten by `k3c daemon` at start-up. this isn't how things will be forever but it is how they are for now.

given your original issue title of "ZFS Support" I assume that you don't care, per se, about pointing k3c at an existing containerd and really just want some way to run k3c with support for the zfs snapshotter. if this is an incorrect assumption please let me know.

**Comment by dweomer** [comment]
@gabeduke you may have some luck trying to pass a custom config via `k3c daemon --config` but the embedded buildkit setup was barfing when passing a copy of `/var/lib/rancher/k3c/etc/containerd/config.toml`

**Comment by gabeduke** [comment]
Thanks for the response @dweomer . I realized after writing this issue that the default containerd config was not being used. I was unable to find exactly how to pass an alternative snapshotter plugin by reading the code (or if it was even possible).

You correctly assume I was just looking for a way to run this on a zfs filesystem. I was able to play around a little bit by running in a VM, but it would be nice to run this on my host system.
